### PR TITLE
Adding unit test workflow

### PR DESCRIPTION
**Pull request by spau6**
Adding unit test workflow

### CODE REVIEW

**Review by amar-c8y — DISMISSED** [review]

**Review by dchc01 — APPROVED** [review]

File: .github/workflows/unit-test-widget.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+env:
+  WORK_DIRECTORY: "cumulocity-asset-properties-widget"
+
+jobs:
+  unit-tests:
+    runs-on: [self-hosted, ghiotapps]
+    if: ${{inputs.skip_run != true }}
+    defaults:
+      run:
+        working-directory: ${{ env.WORK_DIRECTORY }}
```
Comment (ghost) [inline .github/workflows/unit-test-widget.yml]: no need to specify working-directory if it is same as the workspace (repo)

File: .github/workflows/widget-ci.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+on:
+  workflow_dispatch:
+
+  pull_request:
+    paths:
+      - cumulocity-asset-properties-widget/asset-properties-widget/**
+    branches:
+      - develop
+
+env:
```
Comment (ghost) [inline .github/workflows/widget-ci.yml]: remove work directory as it isn't used.

File: .github/workflows/unit-test-widget.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+name: Generic widget unit test
+
+on:
+  workflow_call:
+    inputs:
+      test_command:
+        description: "Command to run widget unit tests. For example: yarn test"
+        required: true
+        type: string
+      skip_run:
```
Comment (ghost) [inline .github/workflows/unit-test-widget.yml:10]: add a skip file for the workflow to run when files that are not in the path provided in the ci yaml file are modified.

**Review by ghost — CHANGES_REQUESTED** [review]

File: .github/workflows/unit-test-widget.yml (hunk 1 above)
Comment (spau6) [reply .github/workflows/unit-test-widget.yml]: removed it

File: .github/workflows/widget-ci.yml (hunk 2 above)
Comment (spau6) [reply .github/workflows/widget-ci.yml]: removed

File: .github/workflows/unit-test-widget.yml (hunk 3 above)
Comment (spau6) [reply .github/workflows/unit-test-widget.yml:10]: Added

**Comment by spau6** [comment]
I have not tested these workflow as it is required to have these workflows in develop branch first. Once it is in develop I may get the option to run them.

**Review by ghost — APPROVED** [review]

**Review by amar-c8y — APPROVED** [review]

**Pull request merged**